plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -965,6 +1015,25 @@ public void executeTableExecute(ConnectorSession session, ConnectorTableExecuteH
         }
     }
 
+    private void executeDropExtendedStats(ConnectorSession session, IcebergTableExecuteHandle executeHandle)
+    {
+        checkArgument(executeHandle.getProcedureHandle() instanceof IcebergDropExtendedStatsHandle);
+
+        Table icebergTable = catalog.loadTable(session, executeHandle.getSchemaTableName());
+
+        verify(transaction == null, "transaction already set");
+        transaction = icebergTable.newTransaction();
+        if (transaction.table().currentSnapshot() != null && transaction.table().currentSnapshot().statisticsFiles() != null) {
+            UpdateTableStatistics updateStatistics = transaction.newUpdateTableStatistics();
+            for (StatisticsFile statisticsFile : transaction.table().currentSnapshot().statisticsFiles()) {
+                updateStatistics.deleteStatisticsFile(statisticsFile.location());
```

Comment:
Probably follow-up, but I could see a use case where you'd want to delete some of the stats attached to a file, but not necessarily all of them.